### PR TITLE
Adding a tutorial on CyclicalLearningRate

### PR DESCRIPTION
**Pull request by sayakpaul**
# Description

Brief Description of the PR:

This PR adds a tutorial on [`CyclicalLearningRate`](https://www.tensorflow.org/addons/api_docs/python/tfa/optimizers/CyclicalLearningRate). 

Fixes # (issue)

## Type of change

- [ ] Bug fix
- [x] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Activation and the changes conform to the [activation contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/activations/README.md#contribution-guidelines)
- [ ] New Callback and the changes conform to the [callback contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/README.md#contribution-guidelines)
- [ ] New Image addition and the changes conform to the [image op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/image/README.md#contribution-guidelines)
- [ ] New Layer and the changes conform to the [layer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/layers/README.md#contribution-guidelines)
- [ ] New Loss and the changes conform to the [loss contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/README.md#contribution-guidelines)
- [ ] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)
- [ ] New Optimizer and the changes conform to the [optimizer contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/optimizers/README.md#contribution-guidelines)
- [ ] New RNN Cell and the changes conform to the [rnn contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/rnn/README.md#contribution-guidelines)
- [ ] New Seq2seq addition and the changes conform to the [seq2seq contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/seq2seq/README.md#contribution-guidelines)
- [ ] New Text addition and the changes conform to the [text op contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/text/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [ ] By running Black + Flake8
    - [ ] By running pre-commit hooks
- [ ] This PR addresses an already submitted issue for TensorFlow Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] This PR contains modifications to C++ custom-ops

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Comment by sayakpaul** [comment]
Regarding code-styling, I have tried adhering to what's instructed [here](https://github.com/tensorflow/addons/blob/master/docs/tutorials/_template.ipynb).

**Comment by sayakpaul** [comment]
@seanpmorgan do you mind taking a look when you get a moment? Many thanks.

**Comment by sayakpaul** [comment]
I think I have added the right links but [the logs](https://github.com/tensorflow/addons/runs/2487633622?check_suite_focus=true) do not suggest so:

image

Anything I might have missed?

**Comment by seanpmorgan** [comment]
@sayakpaul hmmm I don't really see the issues with the links either. Can you try running this to pass the second test and perhaps something with spacing will fix the regex:
```
Please install `nbfmt` and format:
$ python3 -m pip install -U --user git+https://github.com/tensorflow/docs
$ python3 -m tensorflow_docs.tools.nbfmt notebook.ipynb
```

**Comment by sayakpaul** [comment]
@seanpmorgan I actually ran it before creating the PR. That makes it even more surprising I guess.

**Comment by seanpmorgan** [comment]
@googlebot I consent.

**Comment by sayakpaul** [comment]
Thank you, @seanpmorgan. Appreciate the help.

**Comment by sayakpaul** [comment]
@MarkDaoust a gentle reminder on this. If you could help us point in the right direction, that would be great.

**Comment by seanpmorgan** [comment]
cc @lamberta. If time allows do you mind taking a quick look or letting us know if you've seen this before

**Comment by sayakpaul** [comment]
@seanpmorgan on a related note, apart from the build failure, is there anything I can do to improve/modify the tutorial notebook?

**Comment by lamberta** [comment]
Looks new to me. At least I haven't seen this on tf.org and don't find anything comparable on keras.io

File: docs/tutorials/optimizers_cyclicallearningrate.ipynb [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "wg0JjLwH2RTl"
+      },
+      "outputs": [],
+      "source": [
+        "(fig, ax) = plt.subplots(1, 2)\n",
```
Comment (MarkDaoust) [inline docs/tutorials/optimizers_cyclicallearningrate.ipynb:406]: It would be good for this plot of the two subplots had the same `ylim`. Also the lower y-lim should be 0.

File: docs/tutorials/optimizers_cyclicallearningrate.ipynb [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+      "metadata": {
+        "id": "ne0b8aGNyc3v"
+      },
+      "outputs": [],
+      "source": [
+        "steps_per_epoch = len(x_train) // BATCH_SIZE\n",
+        "clr = tfa.optimizers.CyclicalLearningRate(initial_learning_rate=INIT_LR,\n",
+        "    maximal_learning_rate=MAX_LR,\n",
+        "    scale_fn=lambda x: 1/(2.**(x-1)),\n",
+        "    step_size=8 * steps_per_epoch\n",
```
Comment (MarkDaoust) [inline docs/tutorials/optimizers_cyclicallearningrate.ipynb:282]: > 8 * steps_per_epoch

Maybe lower this 8 -> 2 or something  so that the example training task has a chance to run a couple of cycles (or just train a little longer).

**Review by MarkDaoust — CHANGES_REQUESTED** [review]
Overall LGTM.

Attached are the lint fix and a couple of minor suggestions.

File: docs/tutorials/optimizers_cyclicallearningrate.ipynb [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+        "# TensorFlow Addons Optimizers: CyclicalLearningRate"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "MfBg1C5NB3X0"
+      },
+      "source": [
+        "<table class=\"tfa-notebook-buttons\" align=\"left\">\n",
```
[the remark this answers is not in the document]
Comment (sayakpaul) [reply docs/tutorials/optimizers_cyclicallearningrate.ipynb]: Done.

**Comment by sayakpaul** [comment]
Is there anything I can do to for progressing the PR?

**Review by MarkDaoust — DISMISSED** [review]

**Comment by bhack** [comment]
We have a linting failure.

**Comment by sayakpaul** [comment]
@bhack I ran the Docs formatter and pushed the new notebook.

**Review by bhack — APPROVED** [review]

**Comment by sayakpaul** [comment]
Thanks, @bhack. Approximately by when the tutorial will be live? Asking this because it's my first time.

**Comment by sayakpaul** [comment]
@bhack is there anything we could do to fix the build failure?

**Comment by MarkDaoust** [comment]
This isn't automated. I'll run the update.

**Comment by sayakpaul** [comment]
I am sorry. First timer :(

**Comment by MarkDaoust** [comment]
No worries Sayak! 

Thanks for submitting the tutorial. It's published:

https://tensorflow.org/addons/tutorials/optimizers_cyclicallearningrate
